index.ts

```diff
@@ -20,64 +21,83 @@ app.use(express.json());
 
 app.post("/webhook", async (req, res) => {
   console.log("req receieved");
-  const signature = req.headers["x-hub-signature"];
+  const signature = req.headers["x-hub-signature"] as string;
+  console.log(signature)
+  console.log("header",req.headers)
   const payload = JSON.stringify(req.body);
 
-  const hmac = crypto.createHmac("sha1", process.env.GITHUB_SECRET);
+  const hmac = crypto.createHmac("sha1", secret);
   const calculatedSignature = `sha1=${hmac.update(payload).digest("hex")}`;
+  console.log("cals", calculatedSignature)
 
   if (crypto.timingSafeEqual(Buffer.from(signature), Buffer.from(calculatedSignature))) {
-    const { result, respMessage } = await getBranchStatus();
+    const {result, respMessage } = await getBranchStatus(req.body);
     console.log("Result: ", result);
-    res.status(result).send(respMessage);
+    res.status(200).send({ result, respMessage });
   } else {
     res.status(400).send("Invalid Github signature");
   }
 });
 
-app.listen(process.env.PORT, () => {
+interface BranchStatus {
+  result:  number | string;
+  respMessage: string ;
+}
+
+app.listen(port, () => {
   console.log(`Server listening on port ${port}`);
 });
 
-const executeCmd = async (cmd) => {
+const executeCmd = async (cmd: string) => {
   try {
-    const { stdout, stderr } = await exec(cmd);
+    const {stdout, stderr} = await exec(cmd);
     return stderr + "\n" + stdout;
-  } catch (error) {
+  } catch (error: any) {
     console.error(`exec error: ${error}`);
-    throw new Error(stderr + "\n" + stdout);
+    throw new Error(error.stderr + "\n" + error.stdout);
   }
 };
 
-const getBranchStatus = async (req) => {
+async function getBranchStatus(req: any): Promise<BranchStatus> {
```

Comment:
no need for the function keyword, any is forbidden

Reply:
removed function keyword and any